packages/stateful/voting-module-adapter/adapters/CwdVotingCw721Staked/components/ProfileCardMemberInfo/StakingModal.tsx

```diff
@@ -0,0 +1,481 @@
+import { useWallet } from '@noahsaso/cosmodal'
+import { useCallback, useEffect, useState } from 'react'
+import toast from 'react-hot-toast'
+import { useTranslation } from 'react-i18next'
+import {
+  constSelector,
+  useRecoilState,
+  useRecoilValue,
+  useSetRecoilState,
+} from 'recoil'
+
+import {
+  CwdVotingCw721StakedSelectors,
+  refreshDaoVotingPowerAtom,
+  stakingLoadingAtom,
+} from '@dao-dao/state'
+import {
+  ModalLoader,
+  NftSelectionModal,
+  StakingMode,
+  useCachedLoadable,
+} from '@dao-dao/stateless'
+import {
+  BaseStakingModalProps,
+  LoadingDataWithError,
+  NftCardInfo,
+} from '@dao-dao/types'
+import { processError } from '@dao-dao/utils'
+
+import { SuspenseLoader } from '../../../../../components'
+import {
+  Cw721BaseHooks,
+  CwdVotingCw721StakedHooks,
+  useAwaitNextBlock,
+  useWalletProfile,
+} from '../../../../../hooks'
+import { useGovernanceTokenInfo, useStakingInfo } from '../../hooks'
+
+export const StakingModal = (props: BaseStakingModalProps) => (
+  <SuspenseLoader fallback={<ModalLoader onClose={props.onClose} />}>
+    <InnerStakingModal {...props} />
+  </SuspenseLoader>
+)
+
+const InnerStakingModal = ({
+  initialMode = StakingMode.Stake,
+  onClose,
+}: BaseStakingModalProps) => {
+  const { t } = useTranslation()
+  const { address: walletAddress, connected } = useWallet()
+  const { refreshBalances } = useWalletProfile()
+
+  const [actionLabel, setActionLabel] = useState(t('button.stake') as string)
+  const [header, setHeader] = useState<{ title: string; subtitle?: string }>({
```

Comment:
is there a reason to store this display info in state variables? i think the logic that creates these can just be inlined in the component props at the bottom. adding state variables and callbacks to update them adds unnecessary renders and makes the UI slower/use more memory.